### PR TITLE
Update to MVC 5.2.3

### PR DESCRIPTION
**Pull request by cnblogs-dudu**
Follow #2345, Update to MVC 5.2.3.

### CODE REVIEW

**Comment by yishaigalatzer** [comment]
Thanks for a super quick turn around!

@xavierdecoster is going to review in details. Please squash your commits and make sure this update is only about MVC 5.2.3 and no other issues are sneaking in.

**Comment by cnblogs-dudu** [comment]
@yishaigalatzer, I have updated all nuget packages except System.Net.Http and WebGrease.

File: src/NuGetGallery/Web.config [hunk 1]
```diff
@@ -7,426 +7,437 @@
...
+    <!-- ASP.Net settings. -->
+    <!-- ***************** -->
+    <!-- These should never need to be changed -->
+    <add key="webpages:Version" value="3.0.0.0" />
+    <add key="PreserveLoginUrl" value="true" />
+    <add key="ClientValidationEnabled" value="true" />
+    <add key="UnobtrusiveJavaScriptEnabled" value="true" />
+  </appSettings>
+  <connectionStrings>
+    <add name="Gallery.SqlServer" connectionString="Data Source=.;Initial Catalog=NuGetGallery;user id=sa;password=YouChengCNBlogs" providerName="System.Data.SqlClient" />
```
Comment (xavierdecoster) [inline src/NuGetGallery/Web.config]: You might want to change your password :)
Can you please leave the connectionString attribute empty in the code base?

File: src/NuGetGallery/packages.config [hunk 2]
```diff
@@ -2,68 +2,79 @@
...
-  <package id="Glimpse.AspNet" version="1.6.0" targetFramework="net45" />
-  <package id="Glimpse.EF43" version="1.6.0" targetFramework="net45" />
-  <package id="Glimpse.Mvc4" version="1.5.1" targetFramework="net45" />
-  <package id="jQuery" version="1.11.0" targetFramework="net45" />
-  <package id="jQuery.UI.Combined" version="1.10.3" targetFramework="net45" />
-  <package id="jQuery.Validation" version="1.11.1" targetFramework="net45" />
-  <package id="knockoutjs" version="2.2.1" targetFramework="net45" />
+  <package id="EntityFramework" version="6.1.3" targetFramework="net45" />
+  <package id="FontAwesome" version="4.2.0" targetFramework="net45" />
+  <package id="Glimpse" version="1.8.6" targetFramework="net45" />
```
Comment (xavierdecoster) [inline src/NuGetGallery/packages.config]: Can you avoid adding extra dependencies in this PR?
Glimpse is not part of the MVC5 dependencies.
We could consider this in a separate PR, but let's keep this PR focused on the MVC5 upgrade.

**Comment by xavierdecoster** [comment]
I think this PR is overloaded and we should consider smaller PRs with very specific changes.
I see dependencies being added which are not needed at this point (Glimpse, FontAwesome, ...) and I see many package dependency updates which are not needed to get onto MVC5.

We can consider all of those in separate PRs, but I'd really like this PR to focus on the issue it describes: upgrade to MVC 5.2.3.

For one, EF and Azure SDK upgrades could be separate PRs.

Would you consider forking Master and do the minimal required to upgrade to MVC 5.2.3, without all the additional changes you did?

**Comment by cnblogs-dudu** [comment]
@xavierdecoster I'm doing the separate 'upgrade to MVC 5.2.3' PR.

**Comment by xavierdecoster** [comment]
@cnblogs-dudu Thanks! Looking forward

**Comment by chipitsine** [comment]
I guess, AntiForgeryConfig.SuppressXFrameOptionsHeader must be set to "true" when upgrading from MVC4 --> MVC5 ?

MVC5 set header X-Frame-Options by default http://atrophied.co.uk/read/aspnet-mvc-5-antiforgerytoken-and-x-frame-options-sameorigin

that header is also set in Web.config

**Pull request closed without merging**